### ISSUE TITLE
browser-based Desktop Environment

### ISSUE THREAD

**Issue by fazo96** [ipfs/apps#21]
There's a project called [OS.js](http://os.js.org/) which is a desktop environment with included apps such as a file manager, text editor, image manipulator etc and it can run as a static web app.

It has a file system abstraction and APIs to write apps for it. It could be extended with IPFS, for example:

- use IPFS as a repository (OS.js has native repository support for packages)
- integrate IPFS with the file manager
- use IPFS to save session (they can be then restored from everywhere)

Another cool thing that could be done is a js environment to run node CLI apps inside OS.js, in the browser. But that's not related to IPFS.

Just posting this to see if there's interest :+1: Because the use cases for this kind of thing don't sound very valuable to me

**Comment by pguth** [comment]
:astonished: Wow! I am quite a fan of such endeavours.

**Comment by Red5d** [comment]
That does sound useful! And once the javascript version of IPFS is operational, it could even possibly be a package for OS.js to make it into an IPFS node. (might go along with your "integrate IPFS with the file manager" example).

**Comment by jbenet** [comment]
wow, it loaded wolfenstein 3d. so cool.

this is a fantastic project, i'm very impressed.

if all the stuff is modular, by the way, we can use the file manager straight up

**Comment by RichardLitt** [comment]
> Another cool thing that could be done is a js environment to run node CLI apps inside OS.js, in the browser. But that's not related to IPFS.

Couldn't that be used with Chromebooks to make them run IPFS easier?

**Comment by fazo96** [comment]
@RichardLitt good point, but I don't think it's worth the effort if it's only for that purpose, because when IPFS is able to run in browsers we could just write a Chrome extension to run a (limited I guess) daemon.

**Comment by RichardLitt** [comment]
@fazo96 I'm not so sure. Chromebooks are really useful in the developing world because they are so cheap. Anything that helps IPFS work better in cheaper economies is a major +1 in my book.

**Comment by fazo96** [comment]
@RichardLitt I agree with what you say, but what I meant was that using OS.js is not _the best way_ to bring IPFS to chromebooks. The best way I think would be a Chrome extension which would also be relatively easy to do when IPFS runs in browsers, compared to making the command like version of IPFS.js work in OS.js. I think in case a chrome extension can't be done even a web page would be sufficient to start serving files and apps using IPFS.

TLDR: I agree we should get an IPFS daemon to chromebooks, but it's easier to do without going through OS.js in my opinion

**Comment by RichardLitt** [comment]
Fair points! :)
